kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDbusConnectorTest.java

```diff
@@ -0,0 +1,897 @@
+        			+ "		 * connect\n"
+        			+ "		 * disconnect\n"
+        			+ "		 * crit_protocol_start\n"
+        			+ "		 * crit_protocol_stop\n"
+        			+ "		 * update_connect_params\n"
+        			+ "	software interface modes (can always be added):\n"
+        			+ "	valid interface combinations:\n"
+        			+ "		 * #{ managed } <= 1, #{ P2P-device } <= 1, #{ P2P-client, P2P-GO } <= 1,\n"
+        			+ "		   total <= 3, #channels <= 2\n"
+        			+ "		 * #{ managed } <= 1, #{ AP } <= 1, #{ P2P-client } <= 1, #{ P2P-device } <= 1,\n"
+        			+ "		   total <= 4, #channels <= 1\n"
+        			+ "	Device supports scan flush.\n"
+        			+ "	Device supports randomizing MAC-addr in sched scans.\n"
+        			+ "	max # scan plans: 1\n"
+        			+ "	max scan plan interval: 508\n"
+        			+ "	max scan plan iterations: 0\n"
+        			+ "	Supported TX frame types:\n"
+        			+ "		 * managed: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0\n"
+        			+ "		 * AP: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0\n"
+        			+ "		 * P2P-client: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0\n"
+        			+ "		 * P2P-GO: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0\n"
+        			+ "		 * P2P-device: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0\n"
+        			+ "	Supported RX frame types:\n"
+        			+ "		 * managed: 0x40 0xd0\n"
+        			+ "		 * AP: 0x00 0x20 0x40 0xa0 0xb0 0xc0 0xd0\n"
+        			+ "		 * P2P-client: 0x40 0xd0\n"
+        			+ "		 * P2P-GO: 0x00 0x20 0x40 0xa0 0xb0 0xc0 0xd0\n"
+        			+ "		 * P2P-device: 0x40 0xd0\n"
+        			+ "	Supported extended features:\n"
+        			+ "		* [ CQM_RSSI_LIST ]: multiple CQM_RSSI_THOLD records\n"
+        			+ "		* [ 4WAY_HANDSHAKE_STA_PSK ]: 4-way handshake with PSK in station mode\n"
+        			+ "		* [ 4WAY_HANDSHAKE_STA_1X ]: 4-way handshake with 802.1X in station mode\n"
+        			+ "		* [ DFS_OFFLOAD ]: DFS offload\n"
+        			+ "		* [ 4WAY_HANDSHAKE_AP_PSK ]: AP mode PSK offload support";
+        	ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        	outputStream.write(commandOutput.getBytes());
+        	
+        	ByteArrayOutputStream outputStream2 = new ByteArrayOutputStream();
+        	outputStream2.write(commandOutput2.getBytes());
+        	 
+        	ByteArrayOutputStream outputStream3 = new ByteArrayOutputStream();
+        	outputStream3.write(commandOutput3.getBytes());
+        	
+        	CommandStatus commandStatus = mock(CommandStatus.class);
+        	when(commandStatus.getExitStatus()).thenReturn(exitStatus);
+        	when(commandStatus.getOutputStream()).thenReturn(outputStream).thenReturn(outputStream2).thenReturn(outputStream3);
+        	
+            when(this.commandExecutorService.execute(any())).thenReturn(commandStatus);
```

Comment:
I don't like relying on the call order for returning the expected output.

We know which commands will be executed as you can see [here](https://github.com/eclipse/kura/pull/4415/files#r1124107290).

Update and tie together command with the simulated output.

Reply:
Also: wrong indentation

Reply:
I went this route, to avoid mocking the command class as matching input. Will try.

Reply:
It shouldn't be too hard. I think I did something similar in the `triton.ai` tests. Try and take a look there.

Reply:
This took me quite a while but I figured it out! Tiny little thing was causing me headaches all day but it has been solved!!!